notebook/autogen_agent_MathChat.ipynb

```diff
@@ -0,0 +1,752 @@
+    "]\n",
+    "```\n",
+    "\n",
+    "You can directly override it if the above function returns an empty list, i.e., it doesn't find the keys in the specified locations."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Construct agents for MathChat\n",
+    "\n",
+    "We start by initialzing the `AssistantAgent` and `MathUserProxyAgent`. The system message needs to be set to \"You are a helpful assistant.\" for MathChat. The detailed instructions are given in the user message. Later we will use the `MathUserProxyAgent.generate_prompt` to combine the instructions and a math problem for an intial prompt to be sent to the LLM assistant."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from flaml.autogen.agent.assistant_agent import AssistantAgent\n",
+    "from flaml.autogen.agent.math_user_proxy_agent import MathUserProxyAgent\n",
+    "\n",
+    "oai.ChatCompletion.start_logging()\n",
+    "\n",
+    "# 1. create an assistant which is essentially a AssistantAgent instance named \"coding_agent\"\n",
+    "assistant = AssistantAgent(\"assistant\", \n",
+    "                           system_message=\"You are a helpful assistant.\",\n",
+    "                           request_timeout=600, \n",
+    "                           seed=42, \n",
+    "                           config_list=config_list)\n",
+    "\n",
+    "# 2. create the MathUserProxyAgent instance named \"mathproxyagent\"\n",
+    "# By default, the human_input_mode is \"NEVER\", which means the agent will not ask for human input.\n",
+    "mathproxyagent = MathUserProxyAgent(name=\"MathChatAgent\", \n",
+    "                                    human_input_mode=\"NEVER\",\n",
+    "                                    use_docker=False)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Example 1\n",
+    "\n",
+    "Problem: Find all $x$ that satisfy the inequality $(2x+10)(x+3)<(3x+9)(x+8)$. Express your answer in interval notation.\n",
+    "\n",
+    "Solution: \n",
+    "We have \\begin{align*} (2x+10)(x+3)&<(3x+9)(x+8) \\quad \\Rightarrow\n",
+    "\\\\ 2(x+5)(x+3)&<3(x+3)(x+8) \\quad \\Rightarrow\n",
+    "\\\\ 2(x+5)(x+3)-3(x+3)(x+8)&<0 \\quad \\Rightarrow\n",
+    "\\\\ (2x+10-(3x+24))(x+3)&<0 \\quad \\Rightarrow\n",
+    "\\\\ (-x-14)(x+3)&<0 \\quad \\Rightarrow\n",
+    "\\\\ (x+14)(x+3)&>0.\n",
+    "\\end{align*} This inequality is satisfied if and only if $(x+14)$ and $(x+3)$ are either both positive or both negative.  Both factors are positive for $x>-3$ and both factors are negative for $x<-14$.  When $-14<x<-3$, one factor is positive and the other negative, so their product is negative.   Therefore, the range of $x$ that satisfies the inequality is $ \\boxed{(-\\infty, -14)\\cup(-3,\\infty)} $."
```

Comment:
Is it better to print the solution after the problem is solved? To verify the correctness of the solution.

Reply:
Yes, I moved the solution to the end.

Reply:
I tried move it to the end, but I think it looks better to give the solution first. Besides, the correct solution is visualized with box, so it is easy to remember and verify that MathChat is correct.